app/models/manageiq/providers/ansible_tower/automation_manager/template_runner.rb

```diff
@@ -0,0 +1,145 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::TemplateRunner < ::Job
+  DEFAULT_EXECUTION_TTL = 10 # minutes
+
+  # options are job table columns, including options column which is the playbook context info
+  def self.create_job(options)
+    super(name, options.with_indifferent_access)
+  end
+
+  def minimize_indirect
+    @minimize_indirect = true if @minimize_indirect.nil?
+    @minimize_indirect
+  end
+
+  def current_job_timeout(_timeout_adjustment = 1)
+    @execution_ttl ||=
+      (options[:execution_ttl].present? ? options[:execution_ttl].try(:to_i) : DEFAULT_EXECUTION_TTL) * 60
+  end
+
+  def start
+    time = Time.zone.now
+    update_attributes(:started_on => time)
+    miq_task.update_attributes(:started_on => time)
+    my_signal(false, :launch_ansible_tower_job)
```

Comment:
From now on we need to make sure both launching and checking status run on the correct zone, even correct appliance through queue.